Tests/MountebankSwiftTests/Api/Mountebank+Completion.swift

```diff
@@ -0,0 +1,359 @@
+import Foundation
+import XCTest
+@testable import MountebankSwift
+
+extension MountebankTests {
+
```

Comment:
Can we think of something so that we don't have to maintain two identical tests?

Reply:
Perhaps something like: 

```
func testGetImposterAsync() async throws {
    try await testGetImposter(async: true)
}

func testGetImposterCompletion() async throws {
    try await testGetImposter(async: false)
}

func testGetImposter(async: Bool, line: Int = #line) async throws {
    let imposter = Imposter.Examples.simple
    let port = try XCTUnwrap(imposter.value.port)
    let imposterData = try makeDataFromJSON(imposter.json)
    httpClientSpy.httpRequestReturnValue = HTTPResponse(body: imposterData, statusCode: .accepted)

    let assertResult = { (result: Imposter) in
        XCTAssertEqual(imposter.value, result)
    }

    if async {
        assertResult(try await sut.getImposter(port: port))
    } else {
        let completionExpectation = expectation(description: "Expects completion")
        let result: Imposter
        sut.getImposter(port: port) { result in
            switch result {
            case .success(let success):
                assertResult(success)
            case .failure(let error):
                XCTFail("Did expect success but recieved \(error.localizedDescription)")
            }
            completionExpectation.fulfill()
        }

        await fulfillment(of: [completionExpectation], timeout: 1)
    }

    XCTAssertTrue(httpClientSpy.httpRequestCalled)
    XCTAssertEqual(
        httpClientSpy.httpRequestReceivedRequest,
        HTTPRequest(
            url: Endpoint
                .getImposter(
                    port: port,
                    parameters: ImposterParameters(replayable: false, removeProxies: false)
                )
                .makeEndpointUrl(baseUrl: sut.mountebankURL),
            method: .get,
            headers: [HTTPHeaders.contentType: MimeType.json.rawValue]
        )
    )
}
```